pyface/action/action_item.py

```diff
@@ -72,11 +72,11 @@ def _enabled_changed(self, trait_name, old, new):
     def _visible_changed(self, trait_name, old, new):
         self.action.visible = new
 
-    @on_trait_change("_wrappers.control")
-    def _on_destroy(self, object, name, old, new):
+    @observe("_wrappers:items:control")
+    def _on_destroy(self, event):
         """ Handle the destruction of the wrapper. """
-        if name == "control" and new is None:
-            self._wrappers.remove(object)
+        if event.new is None:
```

Comment:
there are a decent number of cases like this, where previously we used `.` not `:` but then only do something when the second trait is the one changed?  

With these types of changes the tests were all still passing and I believe behavior is preserved (code is cleaner IMO as well), but if we want to play it safe or if a reviewer sees why these might not be equivalent, I am happy to adjust